federation/pkg/federation-controller/service/servicecontroller_test.go

```
@@ -17,22 +17,42 @@ limitations under the License.
 package service
 
 import (
+	"sync"
 	"testing"
 
 	"k8s.io/kubernetes/federation/apis/federation"
+	"k8s.io/kubernetes/federation/pkg/dnsprovider/providers/google/clouddns" // Only for unit testing purposes.
 	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/sets"
 )
 
 func TestGetClusterConditionPredicate(t *testing.T) {
+	fakedns, _ := clouddns.NewFakeInterface()
```

Why ignore error again?